app/controllers/repositories_controller.rb

```diff
@@ -0,0 +1,101 @@
+class RepositoriesController < ApplicationController
+  def index
+    @repositories = Repository.where(hide: false).paginate(:page => params[:page])
```

Comment:
What do you think about making `Repository.where(hide: false)` as scope?

Also, check with MAGnificent or SmartRecruit team how to make common logic for pagination.